tools/spec-gen-sdk/src/automation/logging.ts

```diff
@@ -139,8 +139,8 @@ export const loggerWaitToFinish = async (logger: winston.Logger) => {
   for (const transport of logger.transports) {
     if (transport instanceof winston.transports.File) {
       if (transport.end) {
+          await new Promise<void>((resolve) => logger.on('finish', resolve));
```

Comment:
Can you try with this code:
```typescript
/**
 * Ensures all logs are written to file transports
 */
export const loggerWaitToFinish = async (logger: winston.Logger) => {
  logger.info('Wait for logger transports to complete');
  
  const fileTransports = logger.transports.filter((t) => t instanceof winston.transports.File);

  if (fileTransports.length === 0) {
    return;
  }

  await Promise.all(
    fileTransports.map(transport => 
      new Promise<void>(resolve => {
        transport.once('finish', resolve);
        transport.end();
      })
    )
  );
};
```

Reply:
```NodeError: write after end``` happened and find another issues. Will fix!

Reply:
try this:
```typescript
export const loggerWaitToFinish = async (logger: winston.Logger) => {
  logger.info('Wait for logger transports to complete');
  await Promise.all(
    logger.transports.map(
      (transport) =>
        new Promise<void>((resolve) => {
          if (transport instanceof winston.transports.File && transport.end) {
            transport.on('finish', () => resolve());
            transport.end();
          } else {
            resolve();
          }
        })
    )
  );
};
```

Reply:
not works ~ `NodeError: write after end` happened